junit-jupiter-api/src/main/java/org/junit/jupiter/api/extension/ExtensionContext.java

```diff
@@ -173,12 +176,50 @@
 	 * @return the test instance; never {@code null}
 	 * @throws PreconditionViolationException if the test instance is not present
 	 * in this {@code ExtensionContext}
+	 *
+	 * @see #getRequiredTestInstances()
 	 */
+	@API(status = MAINTAINED, since = "5.4")
 	default Object getRequiredTestInstance() {
 		return Preconditions.notNull(getTestInstance().orElse(null),
 			"Illegal state: required test instance is not present in the current ExtensionContext");
 	}
 
+	/**
+	 * Get the test instances associated with the current test or container,
+	 * if available.
+	 *
+	 * <p>While top-level tests only have a single test instance, nested tests
+	 * have one instance per containing test class.
+	 *
+	 * @return an {@code Optional} containing the test instances; never
+	 * {@code null} but potentially empty
+	 * @see #getRequiredTestInstances()
+	 *
+	 * @since 5.4
+	 */
+	@API(status = MAINTAINED, since = "5.4")
```

Comment:
Do we really want to go with `MAINTAINED` at this point? 

Perhaps `EXPERIMENTAL` would be better suited for this feature set.